### DISCUSSION TITLE
Invite only media CMS with shareable links

### DISCUSSION THREAD

**Discussion by flinthamm** [mediacms-io/mediacms#1177] [Feature Requests]
First of all, thanks for this great project and your hard work :)

**Describe the feature you'd like**
Correct me if I'm wrong but in order to have an invite only CMS, whereby you can only view each others media if logged in, it is necessary to make all uploads `public `and to set the `GLOBAL_LOGIN_REQUIRED = True`. This is fine except if you then want to share any video with the outside world you can't, as it would then require you to create an account and have anyone wanting to view the video to login first? If you change the upload to `unlisted`,  the global login required flag overrides this and still requires a guest to login, despite having a complex link.

**Describe alternatives you've considered**
If `GLOBAL_LOGIN_REQUIRED = False` is set, then in order for other users to see your content the upload will have to be made either `public `(in which case the whole world can see the content) or `private`. I think when `private`, it then isn't visible to other logged in users and certainly doesn't appear in the majority of places but most importantly you then can't share with the outside world. 

Finally if you set an upload as `unlisted`, then you can share with the outside world but only you can see your own media and no other users that login.

The feature I'm suggesting here is in the first scenario (an invite only CMS) it should be possible to still share individual links to the outside world, without them having to login and still allowing logged in users to see your content. I hope this is clear?

**Comment by francescocaponio** [comment]
I too would like to have a behaviour in which me and my family and close friends can navigate through the site while logged in, without login you cannot navigate through the website (GLOBAL_LOGIN_REQUIRED = True), but you still can share a link where even unregistered visits can see the video. In this way, you can choose to upload a private video only for you or something that other users on the same platform can see, knowing that the outside world can't see it, but if you may need to share a video outside, there is still a way to create a shared link.
(the same behaviour you have on immich for instance. They even let you share for a certain amount of time, or with write permission on an album...)
![image](https://github.com/user-attachments/assets/6e81a0da-588f-46b2-a0d2-4a94ba9590b0)
